### PR TITLE
feat(ojoi): OJOI application improvements/refactor

### PR DESCRIPTION
**Pull request by jonbjarnio**
## What

- Updated clients to match changes made to the DMR apis.
- Moved all graphql logic to hooks
- Simplified application structure
- Fixed / improved UI and UX feel of the application
- Added better error handling and error messages.

## Why

Bug fixes and improvements for the application from the first iteration, for better user and developer experience.

## Screenshots / Gifs

image

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced new GraphQL mutations and queries for managing application attachments and comments.
  - Added functionality for generating presigned URLs for secure file uploads.
  - Implemented methods for adding, retrieving, and deleting application attachments.

- **Enhancements**
  - Improved error handling and response structure for application-related operations.
  - Enhanced the `postComment` method to return success status.

- **Bug Fixes**
  - Resolved issues related to data fetching and component rendering.

- **Documentation**
  - Updated documentation to reflect new components and API changes.

### CODE REVIEW

File: libs/api/domains/official-journal-of-iceland-application/src/lib/ojoiApplication.service.ts [hunk 1]
```diff
@@ -42,4 +59,68 @@ export class OfficialJournalOfIcelandApplicationService {
...
+  async deleteApplicationAttachment(input: DeleteApplicationAttachmentInput) {
+    try {
+      await this.ojoiApplicationService.deleteApplicationAttachment({
+        id: input.applicationId,
+        key: input.key,
+      })
+
+      return { success: true }
+    } catch (error) {
+      return { success: false }
```
Comment (thordurhhh) [inline libs/api/domains/official-journal-of-iceland-application/src/lib/ojoiApplication.service.ts:123]: please add logging to errors here and line 101

File: libs/api/domains/official-journal-of-iceland-application/src/models/addApplicationAttachment.input.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { InputType, Field, Int } from '@nestjs/graphql'
+
+@InputType('OfficialJournalOfIcelandApplicationAddApplicationAttachmentInput')
+export class AddApplicationAttachmentInput {
+  @Field(() => String)
+  applicationId!: string
+
+  @Field(() => String)
```
Comment (thordurhhh) [inline libs/api/domains/official-journal-of-iceland-application/src/models/addApplicationAttachment.input.ts:8]:
```suggestion
  @Field()
```

File: libs/api/domains/official-journal-of-iceland-application/src/models/addApplicationAttachment.response.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Field, ObjectType } from '@nestjs/graphql'
+
+@ObjectType(
+  'OfficialJournalOfIcelandApplicationAddApplicationAttachmentResponse',
+)
+export class AddApplicationAttachmentResponse {
+  @Field(() => Boolean)
```
Comment (thordurhhh) [inline libs/api/domains/official-journal-of-iceland-application/src/models/addApplicationAttachment.response.ts:7]:
```suggestion
  @Field()
```

**Review by thordurhhh — APPROVED** [review]
It's a beauty 😍

**Comment by jonbjarnio** [comment]
@coderabbit resolve

**Comment by jonbjarnio** [comment]
@coderabbitai resolve

**Pull request merged**